### PR TITLE
Improving pipeline tests

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

This PR removes `small_models` in favor of random dynamic super small models from tiny_config.

- Main reason, so we can test the pipeline on as much architectures as possible (which is the goal of the pipeline)
- Main drawback, we don't test `fast` vs `non-fast` tokenizers anymore (non fast cannot be learned dynamically on the fly like
the fast tokenizers)
- We cannot test exact values either because the models are random. Introduces ANY help class to make output structure more readable within a test too !
- We don't test explicity TF vs PT outputs on the small tiny config (again random weights), but we will explicitely add those for the slow tests which are going to be much more explicit (with actual values being checked against)

Tl;dr small_models -> random weight with more configs out of the box, large_models, less but check more information within outputs.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@LysandreJik 

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/test_pipelines_common.py [hunk 1]
```diff
@@ -12,15 +12,126 @@
...
+        return module._CHECKPOINT_FOR_DOC
+    else:
+        logger.warning(f"Can't retrieve checkpoint from {architecture.__name__}")
+
+
+def get_tiny_config_from_class(configuration_class):
+    if "OpenAIGPT" in configuration_class.__name__:
+        # This is the only file that is inconsistent with the naming scheme.
+        # Will rename this file if we decide this is the way to go
+        return
```
Comment (LysandreJik) [inline tests/test_pipelines_common.py:44]: Will need to handle this

**Review by LysandreJik — APPROVED** [review]
Great, LGTM! Let's keep an eye on the execution times of the pipeline tests; if they become too large, we'll have to think about reducing it - but otherwise, looks great.

**Comment by Narsil** [comment]
So let's start to merge those iteratively ?

**Comment by LysandreJik** [comment]
Yes :)
